pex/resolve/resolver_options.py

```diff
@@ -564,13 +565,15 @@ def create_network_configuration(options):
 
     :param options: The Pip resolver configuration options.
     """
-    return NetworkConfiguration(
+    network_configuration = NetworkConfiguration(
         retries=options.retries,
         timeout=options.timeout,
         proxy=options.proxy,
         cert=options.cert,
         client_cert=options.client_cert,
     )
+    initialize_ssl_context(network_configuration=network_configuration)
```

Comment:
I'm not super happy with this back door way of threading in an SSLContext from the main thread, but there are alot of paths / layers as it stands right now and this is solid, since the configuration was centralized a good while ago and it unbreaks users now leaving room for improvement (explicit threading of an SSLContext from main on down) later.